src/test/java/org/janelia/saalfeldlab/n5/AbstractN5Test.java

```diff
@@ -509,4 +512,55 @@ public void testVersion() throws NumberFormatException, IOException {
 
 		Assert.assertFalse(N5Reader.VERSION.isCompatible(n5.getVersion()));
 	}
+
+	@Test
+	public void testDelete() throws IOException {
+		final String datasetName = AbstractN5Test.datasetName + "-test-delete";
+		n5.createDataset(datasetName, dimensions, blockSize, DataType.UINT8, new RawCompression());
+		final DatasetAttributes attributes = n5.getDatasetAttributes(datasetName);
+		final long[] position1 = {0, 0, 0};
+		final long[] position2 = {0, 1, 2};
+
+		// no blocks should exist to begin with
+		Assert.assertTrue(isNullOrBytesAndAllZeros(n5.readBlock(datasetName, attributes, position1)));
+		Assert.assertTrue(isNullOrBytesAndAllZeros(n5.readBlock(datasetName, attributes, position2)));
+
+		final ByteArrayDataBlock dataBlock = new ByteArrayDataBlock(blockSize, position1, byteBlock);
+		n5.writeBlock(datasetName, attributes, dataBlock);
+
+		// block should exist at position1 but not at position2
+		final DataBlock<?> readBlock = n5.readBlock(datasetName, attributes, position1);
+		Assert.assertNotNull(readBlock);
+		Assert.assertThat(readBlock, IsInstanceOf.instanceOf(ByteArrayDataBlock.class));
+		Assert.assertArrayEquals(byteBlock, ((ByteArrayDataBlock) readBlock).getData());
+		Assert.assertTrue(isNullOrBytesAndAllZeros(n5.readBlock(datasetName, attributes, position2)));
+
+		// deletion should report true only at first deletion of position1
+		Assert.assertTrue(n5.deleteBlock(datasetName, position1));
+		Assert.assertTrue(n5.deleteBlock(datasetName, position1));
+		Assert.assertTrue(n5.deleteBlock(datasetName, position2));
+
+		// no block should exist anymore
+		Assert.assertTrue(isNullOrBytesAndAllZeros(n5.readBlock(datasetName, attributes, position1)));
+		Assert.assertTrue(isNullOrBytesAndAllZeros(n5.readBlock(datasetName, attributes, position2)));
+	}
+
+	// Certain N5 implementations (HDF5) do not necessarily store data as blocks/chunks.
+	// For this test, assume that a non-existing block is returned as all zeros in such
+	// a scenario.
+	private boolean isNullOrBytesAndAllZeros(final DataBlock<?> dataBlock) {
```

Comment:
I think this should be an override in n5-hdf5's test?

Reply:
I made the method `protected abstract` and each N5 implementation will have to implement this method according to what it means that a block is deleted.

Reply:
Not a big thing, but this null check could be the default implementation -- that way the tests in cloud implementations will work without adding anything extra, since they are consistent with returning `null` for non-existing blocks.

Reply:
I thought about making it default but decided against it. In this case, I think being explicit is better because (other) downstream implementations will have to implement the new interface anyways and will not run into unexpected test failures when this method is not also overridden.